library/core/src/ptr/const_ptr.rs

```diff
@@ -1774,3 +1792,20 @@ impl<T: ?Sized> PartialOrd for *const T {
         *self >= *other
     }
 }
+
+// This module contains all proof harnesses for function contracts.
+// Each proof harness verifies the soundness of a function contract for a specific type.
+#[cfg(kani)]
+#[unstable(feature = "kani", issue = "none")]
+mod verify {
+    use crate::kani;
+
+    #[allow(unused)]
+    #[kani::proof_for_contract(<*const i32>::offset)]
+    fn check_offset_i32() {
+        let mut test_val: i32 = kani::any();
+        let test_ptr: *const i32 = &test_val;
+        let offset: isize = kani::any();
+        unsafe { test_ptr.offset(offset) };
+    }
+}
```

Comment:
We can expand this to other types. Please, see my [comment here](https://github.com/model-checking/verify-rust-std/pull/91#discussion_r1778937098) for clarity.

Reply:
Thank you! We will implement that for other types.